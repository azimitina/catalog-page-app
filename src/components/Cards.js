import React from "react";
import { Card } from "semantic-ui-react";
import "./Cards.scss";

const Cards = ({
  name,
  meta,
  price,
  final_price,
  marketingMessaging,
  image,
}) => (
  <Card
    image={image}
    header={name}
    meta={meta}
    description={
      <div>
        <span className={price > final_price ? "price-line-through" : ""}>
          ${price}
        </span>
        {price > final_price && (
          <span
            style={{ marginLeft: "5px", color: "#FF0000" }}
          >{`$${final_price}`}</span>
        )}
        <h6 className="marketing-msg">{marketingMessaging}</h6>
      </div>
    }
  />
);

export default Cards;
