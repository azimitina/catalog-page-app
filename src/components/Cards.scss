.ui.card {
  text-align: left;

  & .content {
    height: 120px;

    & .meta,
    & .header {
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }

    & .marketing-msg {
      color: #008000;
      margin: 10px 0 0 0;
    }

    & .price-line-through {
      text-decoration: line-through;
      color: #a9a9a9;
    }
  }
}
