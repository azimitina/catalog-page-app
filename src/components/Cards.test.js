import { shallow } from "enzyme";
import Cards from "./Cards";

describe("Cards", () => {
  it("should render Card", () => {
    const props = {
      name: "Peppy Co",
      meta: "LED Light",
      price: "$180",
      final_price: "$180",
      image: "http://example.com/fancy.png",
    };
    const wrapper = shallow(<Cards {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render Card with marketing message", () => {
    const props = {
      name: "Peppy Co",
      meta: "LED Light",
      price: "$180",
      final_price: "$180",
      image: "http://example.com/fancy.png",
      marketingMessaging: "Gift without the risk",
    };
    const wrapper = shallow(<Cards {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
