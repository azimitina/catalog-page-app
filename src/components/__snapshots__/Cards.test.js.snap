// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards should render Card 1`] = `
<Card
  description={
    <div>
      <span
        className=""
      >
        $
        $180
      </span>
      <h6
        className="marketing-msg"
      />
    </div>
  }
  header="Peppy Co"
  image="http://example.com/fancy.png"
  meta="LED Light"
/>
`;

exports[`Cards should render Card with marketing message 1`] = `
<Card
  description={
    <div>
      <span
        className=""
      >
        $
        $180
      </span>
      <h6
        className="marketing-msg"
      >
        Gift without the risk
      </h6>
    </div>
  }
  header="Peppy Co"
  image="http://example.com/fancy.png"
  meta="LED Light"
/>
`;
